ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -656,33 +656,205 @@ let text_document_lens (state : State.t)
+          push range;
+          let module_expr = module_binding.pmb_expr in
+          iterator.module_expr iterator module_expr
+        | Pstr_modtype module_type_declaration ->
+          iterator.module_type_declaration iterator module_type_declaration
+        | Pstr_type (_, type_declarations) ->
+          List.iter type_declarations ~f:(fun type_declaration ->
+              iterator.type_declaration iterator type_declaration)
+        | Parsetree.Pstr_eval (expr, _) -> iterator.expr iterator expr
+        | Parsetree.Pstr_primitive _ -> ()
+        | Parsetree.Pstr_typext _ -> ()
+        | Parsetree.Pstr_exception _ -> ()
+        | Parsetree.Pstr_recmodule _ -> ()
+        | Parsetree.Pstr_open _ -> ()
+        | Parsetree.Pstr_class class_declarations ->
+          List.iter class_declarations ~f:(fun class_declaration ->
+              iterator.class_declaration iterator class_declaration)
+        | Parsetree.Pstr_class_type _ -> ()
+        | Parsetree.Pstr_include _ -> ()
+        | Parsetree.Pstr_attribute _ -> ()
+        | Parsetree.Pstr_extension (_, _) -> ()
+      in
+      let signature_item (iterator : Ast_iterator.iterator)
+          (signature_item : Parsetree.signature_item) =
+        match signature_item.psig_desc with
+        | Psig_value _ -> ()
+        | Psig_type (_, type_declarations) ->
+          List.iter type_declarations ~f:(fun type_declaration ->
+              iterator.type_declaration iterator type_declaration)
+        | Psig_typesubst _ -> ()
+        | Psig_typext _ -> ()
+        | Psig_exception _ -> ()
+        | Psig_module module_declaration ->
+          iterator.module_declaration iterator module_declaration
+        | Psig_modsubst _ -> ()
+        | Psig_recmodule _ -> ()
+        | Psig_modtype module_type_declaration ->
+          iterator.module_type_declaration iterator module_type_declaration
+        | Psig_modtypesubst _ -> ()
+        | Psig_open _ -> ()
+        | Psig_include _ -> ()
+        | Psig_class _ -> ()
+        | Psig_class_type _ -> ()
+        | Psig_attribute _ -> ()
+        | Psig_extension (_, _) -> ()
+      in
+      { default_iterator with
+        class_declaration
+      ; expr
+      ; module_declaration
+      ; module_type
+      ; module_type_declaration
+      ; signature
+      ; signature_item
+      ; structure
+      ; structure_item
+      ; type_declaration
+      ; value_binding
+      }
     in
-    loop [] outline
-    |> List.sort ~compare:(fun x y -> Ordering.of_int (compare x y))
+    match parsetree with
+    | `Interface signature -> iterator.signature iterator signature
+    | `Implementation structure -> iterator.structure iterator structure
+  in
+  let doc = Document_store.get state.store uri in
+  let+ () =
+    Document.with_pipeline_exn doc (fun pipeline ->
+        let parsetree = Mpipeline.reader_parsetree pipeline in
+        fold_over_parsetree parsetree)
   in
-  Some folds
+  Some (List.rev !ranges)
```

Comment:
Reverting the list is not necessary. I did it only to keep it in the same order as the previous implementation and avoid noises in test results.